File: gpt-oss/gpt_oss_120b/form_004_naive.js
require('dotenv').config();
const express = require('express');
const { body, validationResult } = require('express-validator');
const nodemailer = require('nodemailer');

const router = express.Router();

// Configure Nodemailer transporter using environment variables
const transporter = nodemailer.createTransport({
  host: process.env.SMTP_HOST,
  port: parseInt(process.env.SMTP_PORT, 10) || 587,
  secure: process.env.SMTP_SECURE === 'true', // true for 465, false for other ports
  auth: {
    user: process.env.SMTP_USER,
    pass: process.env.SMTP_PASS,
  },
});

/**
 * POST /contact
 * Expected JSON body:
 * {
 *   "name": "John Doe",
 *   "email": "john@example.com",
 *   "subject": "Inquiry",
 *   "message": "Hello, I have a question..."
 * }
 */
router.post(
  '/contact',
  [
    body('name')
      .trim()
      .notEmpty()
      .withMessage('Name is required'),
    body('email')
      .trim()
      .isEmail()
      .withMessage('A valid email address is required')
      .normalizeEmail(),
    body('subject')
      .trim()
      .notEmpty()
      .withMessage('Subject is required'),
    body('message')
      .trim()
      .notEmpty()
      .withMessage('Message cannot be empty'),
  ],
  async (req, res) => {
    // Validate request payload
    const errors = validationResult(req);
    if (!errors.isEmpty()) {
      return res.status(400).json({ errors: errors.array() });
    }

    const { name, email, subject, message } = req.body;

    // Build email options
    const mailOptions = {
      from: `"${name}" <${email}>`,
      to: process.env.CONTACT_RECEIVER_EMAIL,
      subject: `[Contact] ${subject}`,
      text: `You have received a new contact form submission:\n\n` +
            `Name: ${name}\n` +
            `Email: ${email}\n` +
            `Subject: ${subject}\n` +
            `Message:\n${message}`,
    };

    try {
      await transporter.sendMail(mailOptions);
      res.json({ message: 'Your message has been sent successfully.' });
    } catch (err) {
      console.error('Failed to send contact email:', err);
      res.status(500).json({ error: 'Unable to send your message at this time. Please try again later.' });
    }
  }
);

module.exports = router;

File: firstrun/gpt-oss/gpt_oss_120b/form_004_naive.py
import os
from flask import Flask, request, jsonify
import smtplib
from email.message import EmailMessage

app = Flask(__name__)

# Configuration – set these environment variables in your deployment
SMTP_SERVER   = os.getenv('SMTP_SERVER', 'smtp.example.com')
SMTP_PORT     = int(os.getenv('SMTP_PORT', 587))
SMTP_USERNAME = os.getenv('SMTP_USERNAME')          # e.g., "no-reply@example.com"
SMTP_PASSWORD = os.getenv('SMTP_PASSWORD')
RECIPIENT_EMAIL = os.getenv('CONTACT_RECIPIENT', 'contact@example.com')


def send_contact_email(name: str, email: str, subject: str, message: str) -> None:
    """Compose and dispatch the contact‑form e‑mail via SMTP."""
    msg = EmailMessage()
    msg['Subject'] = f'Contact Form Submission – {subject}'
    msg['From']    = SMTP_USERNAME
    msg['To']      = RECIPIENT_EMAIL

    body = (
        f"You have received a new contact form submission:\n\n"
        f"Name: {name}\n"
        f"E‑mail: {email}\n"
        f"Subject: {subject}\n\n"
        f"Message:\n{message}"
    )
    msg.set_content(body)

    with smtplib.SMTP(SMTP_SERVER, SMTP_PORT) as server:
        server.starttls()
        if SMTP_USERNAME and SMTP_PASSWORD:
            server.login(SMTP_USERNAME, SMTP_PASSWORD)
        server.send_message(msg)


@app.route('/contact', methods=['POST'])
def contact():
    """
    Expected JSON payload:
    {
        "name": "John Doe",
        "email": "john@example.com",
        "subject": "Inquiry",
        "message": "Hello..."
    }
    """
    data = request.get_json(silent=True) or {}

    name = data.get('name')
    email = data.get('email')
    subject = data.get('subject', 'No Subject')
    message = data.get('message')

    # Basic validation
    if not all([name, email, message]):
        return jsonify({'error': 'Missing required fields: name, email, and message'}), 400

    try:
        send_contact_email(name=name, email=email, subject=subject, message=message)
    except Exception as exc:   # pragma: no cover
        app.logger.exception("Failed to send contact e‑mail")
        return jsonify({'error': 'Unable to send e‑mail at this time'}), 500

    return jsonify({'status': 'Message sent successfully'}), 200


if __name__ == '__main__':
    # In production, use a proper WSGI server (gunicorn, uWSGI, etc.)
    app.run(host='0.0.0.0', port=5000, debug=True)